TTS crash if TTS initialization fails

###### Reproduction Steps

1. Enable text to speech on a phone that doesn't have a TTS engine or is configured to use a TTS engine that can't start properly
2. Preview or review a card
3. 


###### Expected Result

AnkiDroid shows a toast that it is initializing TTS, then a toast that TTS couldn't be initialized


###### Actual Result

Crash

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/69a77947-0a13-42dc-9319-4d97cf4fbea6
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.Reviewer}: java.lang.NullPointerException: Attempt to invoke virtual method 'int android.speech.tts.TextToSpeech.setOnUtteranceProgressListener(android.speech.tts.UtteranceProgressListener)' on a null object reference
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2822)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2897)
at android.app.ActivityThread.-wrap11(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:251)
at android.app.ActivityThread.main(ActivityThread.java:6572)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int android.speech.tts.TextToSpeech.setOnUtteranceProgressListener(android.speech.tts.UtteranceProgressListener)' on a null object reference
at com.ichi2.anki.ReadText$3.onInit(ReadText.java:289)
at android.speech.tts.TextToSpeech.dispatchOnInit(TextToSpeech.java:824)
at android.speech.tts.TextToSpeech.initTts(TextToSpeech.java:802)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:733)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:712)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:696)
at com.ichi2.anki.ReadText.initializeTts(ReadText.java:272)
at com.ichi2.anki.AbstractFlashcardViewer.onCollectionLoaded(AbstractFlashcardViewer.java:871)
at com.ichi2.anki.Reviewer.onCollectionLoaded(Reviewer.java:133)
at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:261)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:827)
at com.ichi2.anki.Reviewer.onCreate(Reviewer.java:72)
at android.app.Activity.performCreate(Activity.java:6975)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2775)
... 9 more
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.speech.tts.TextToSpeech.setOnUtteranceProgressListener(android.speech.tts.UtteranceProgressListener)' on a null object reference
at com.ichi2.anki.ReadText$3.onInit(ReadText.java:289)
at android.speech.tts.TextToSpeech.dispatchOnInit(TextToSpeech.java:824)
at android.speech.tts.TextToSpeech.initTts(TextToSpeech.java:802)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:733)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:712)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:696)
at com.ichi2.anki.ReadText.initializeTts(ReadText.java:272)
at com.ichi2.anki.AbstractFlashcardViewer.onCollectionLoaded(AbstractFlashcardViewer.java:871)
at com.ichi2.anki.Reviewer.onCollectionLoaded(Reviewer.java:133)
at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:261)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:827)
at com.ichi2.anki.Reviewer.onCreate(Reviewer.java:72)
at android.app.Activity.performCreate(Activity.java:6975)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2775)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2897)
at android.app.ActivityThread.-wrap11(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:251)
at android.app.ActivityThread.main(ActivityThread.java:6572)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2897)
at android.app.ActivityThread.-wrap11(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:251)
at android.app.ActivityThread.main(ActivityThread.java:6572)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
at com.ichi2.anki.ReadText$3.onInit(ReadText.java:289)
at android.speech.tts.TextToSpeech.dispatchOnInit(TextToSpeech.java:824)
at android.speech.tts.TextToSpeech.initTts(TextToSpeech.java:802)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:733)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:712)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:696)
at com.ichi2.anki.ReadText.initializeTts(ReadText.java:272)
at com.ichi2.anki.AbstractFlashcardViewer.onCollectionLoaded(AbstractFlashcardViewer.java:871)
at com.ichi2.anki.Reviewer.onCollectionLoaded(Reviewer.java:133)
at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:261)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:827)
at com.ichi2.anki.Reviewer.onCreate(Reviewer.java:72)
at android.app.Activity.performCreate(Activity.java:6975)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2775)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2897)
at android.app.ActivityThread.-wrap11(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:251)
at android.app.ActivityThread.main(ActivityThread.java:6572)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
at com.ichi2.anki.ReadText$3.onInit(ReadText.java:289)
at android.speech.tts.TextToSpeech.dispatchOnInit(TextToSpeech.java:824)
at android.speech.tts.TextToSpeech.initTts(TextToSpeech.java:802)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:733)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:712)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:696)
at com.ichi2.anki.ReadText.initializeTts(ReadText.java:272)
at com.ichi2.anki.AbstractFlashcardViewer.onCollectionLoaded(AbstractFlashcardViewer.java:871)
at com.ichi2.anki.Reviewer.onCollectionLoaded(Reviewer.java:133)
at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:261)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:827)
at com.ichi2.anki.Reviewer.onCreate(Reviewer.java:72)
at android.app.Activity.performCreate(Activity.java:6975)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2775)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2897)
at android.app.ActivityThread.-wrap11(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:251)
at android.app.ActivityThread.main(ActivityThread.java:6572)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
at com.ichi2.anki.ReadText$3.onInit(ReadText.java:289)
at android.speech.tts.TextToSpeech.dispatchOnInit(TextToSpeech.java:824)
at android.speech.tts.TextToSpeech.initTts(TextToSpeech.java:802)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:733)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:712)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:696)
at com.ichi2.anki.ReadText.initializeTts(ReadText.java:272)
at com.ichi2.anki.AbstractFlashcardViewer.onCollectionLoaded(AbstractFlashcardViewer.java:871)
at com.ichi2.anki.Reviewer.onCollectionLoaded(Reviewer.java:133)
at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:261)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:827)
at com.ichi2.anki.Reviewer.onCreate(Reviewer.java:72)
at android.app.Activity.performCreate(Activity.java:6975)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2775)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2897)
at android.app.ActivityThread.-wrap11(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1598)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:251)
at android.app.ActivityThread.main(ActivityThread.java:6572)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
at com.ichi2.anki.ReadText$3.onInit(ReadText.java:289)
at android.speech.tts.TextToSpeech.dispatchOnInit(TextToSpeech.java:824)
at android.speech.tts.TextToSpeech.initTts(TextToSpeech.java:802)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:733)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:712)
at android.speech.tts.TextToSpeech.<init>(TextToSpeech.java:696)
at com.ichi2.anki.ReadText.initializeTts(ReadText.java:272)
at com.ichi2.anki.AbstractFlashcardViewer.onCollectionLoaded(AbstractFlashcardViewer.java:871)
at com.ichi2.anki.Reviewer.onCollectionLoaded(Reviewer.java:133)
at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:261)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:827)
at com.ichi2.anki.Reviewer.onCreate(Reviewer.java:72)
at android.app.Activity.performCreate(Activity.java:6975)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2775)

--- a/AnkiDroid/src/main/java/com/ichi2/anki/ReadText.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/ReadText.java
@@ -283,27 +283,28 @@ public class ReadText {
                         Toast.makeText(mReviewer.get(), mReviewer.get().getString(R.string.no_tts_available_message), Toast.LENGTH_LONG).show();
                         Timber.w("TTS initialized but no available languages found");
                     }
+                    mTts.setOnUtteranceProgressListener(new UtteranceProgressListener() {
+                        @Override
+                        public void onDone(String arg0) {
+                            if (ReadText.sTextQueue.size() > 0) {
+                                String[] text = ReadText.sTextQueue.remove(0);
+                                ReadText.speak(text[0], text[1], TextToSpeech.QUEUE_FLUSH);
+                            }
+                        }
+                        @Override
+                        @Deprecated
+                        public void onError(String arg0) {
+                            // do nothing
+                        }
+                        @Override
+                        public void onStart(String arg0) {
+                            // no nothing
+                        }
+                    });
                 } else {
                     Toast.makeText(mReviewer.get(), mReviewer.get().getString(R.string.no_tts_available_message), Toast.LENGTH_LONG).show();
+                    Timber.w("TTS not successfully initialized");
                 }
-                mTts.setOnUtteranceProgressListener(new UtteranceProgressListener() {
-                    @Override
-                    public void onDone(String arg0) {
-                        if (ReadText.sTextQueue.size() > 0) {
-                            String[] text = ReadText.sTextQueue.remove(0);
-                            ReadText.speak(text[0], text[1], TextToSpeech.QUEUE_FLUSH);
-                        }
-                    }
-                    @Override
-                    @Deprecated
-                    public void onError(String arg0) {
-                        // do nothing
-                    }
-                    @Override
-                    public void onStart(String arg0) {
-                        // no nothing
-                    }
-                });
             }
         });
         mTtsParams = new HashMap<>();

